From 4a2164cb2182b6e7fd551374a6aea85a35469070 Mon Sep 17 00:00:00 2001
From: Heinz Wrobel <Heinz.Wrobel@nxp.com>
Date: Tue, 8 Nov 2022 10:55:16 +0100
Subject: [PATCH 01/20] pseudo: Enabled utimensat/ftimens

This required a makewrappers change to permit array arguments to wrappers.

Note: as of pseudo 1.9.8 the makewrappers script already supports array
arguments natively (self.array handling and the array regex are present
upstream), so the original makewrappers hunk has been dropped to avoid a
patch conflict after the Yocto subminor update. Only the wrapfuncs.in
change that declares the utimensat/futimens wrappers remains necessary.

Upstream-Status: Inappropriate [NXP S32 / ubuntu rootfs support only]
Signed-off-by: Heinz Wrobel <Heinz.Wrobel@nxp.com>
---
 ports/unix/wrapfuncs.in |  2 ++
 1 file changed, 2 insertions(+)

diff --git a/ports/unix/wrapfuncs.in b/ports/unix/wrapfuncs.in
index 7724fc7..4d7a887 100644
--- a/ports/unix/wrapfuncs.in
+++ b/ports/unix/wrapfuncs.in
@@ -29,6 +29,8 @@ char *tmpnam(char *s);
 int truncate(const char *path, off_t length);
 int utime(const char *path, const struct utimbuf *buf);
 int utimes(const char *path, const struct timeval *times);
+int utimensat(int dirfd, const char *path, const struct timespec times[2], int flags);
+int futimens(int fd, const struct timespec times[2]);
 # needed because libc stdio does horrible things with inline asm syscalls
 FILE *fopen(const char *path, const char *mode); /* noignore_path=1 */
 int fclose(FILE *fp);  /* noignore_path=1 */
-- 
2.34.1
